src/Octoshift/GithubClient.cs

```diff
@@ -16,13 +16,19 @@ public class GithubClient
     {
         private readonly HttpClient _httpClient;
         private readonly OctoLogger _log;
+        private double _retryDelay;
         private readonly RetryPolicy _retryPolicy;
+        private readonly DateTimeProvider _dateTimeProvider;
 
-        public GithubClient(OctoLogger log, HttpClient httpClient, IVersionProvider versionProvider, RetryPolicy retryPolicy, string personalAccessToken)
+        private const string DEFAULT_RATE_LIMIT_REMAINING = "5000";
+        private const float MILLISECONDS_PER_SECOND = 1000f;
```

Comment:
This doesn't need to be a `float`:
```suggestion
        private const int MILLISECONDS_PER_SECOND = 1000;
```

Reply:
CodeQL [whined](https://github.com/github/gh-gei/pull/623#pullrequestreview-1106528791) about possible loss of precision when casting the result of integer multiplication to a double, but if I should just leave it an integer and tell CodeQL to take a hike, I will!

Reply:
oh that's because the `_retryDelay` is a `double` but I guess it shouldn't be right?

Reply:
should probably be an int